### PR TITLE
Replace wp-scripts env usage with wp-env in CI.

### PR DESCRIPTION
**Pull request by talldan**
## Description
Uses `wp-env` for running Travis e2e tests instead of `wp-scripts env`.

While the branch is named `deprecate/env` I haven't actually addressed any deprecations in this PR. I think that can be handled separately, I haven't renamed the branch as I think there's some useful history in the PR.

### CODE REVIEW

**Comment by gziolo** [comment]
What about `package.json` file?

https://github.com/WordPress/gutenberg/blob/59b07cc57adff90630fc9d5cf2317269a0f4f158/package.json#L215-L216

https://github.com/WordPress/gutenberg/blob/59b07cc57adff90630fc9d5cf2317269a0f4f158/package.json#L228

@epiqueras - do you have some other open PRs that replace `wp-scripts env` with `wp-env`?

**Comment by epiqueras** [comment]
>@epiqueras - do you have some other open PRs that replace wp-scripts env with wp-env?

Mine are all stale but I am pretty sure @noahtallen plans on doing this after he lands #20090.

**Comment by talldan** [comment]
@gziolo Still testing the Travis config at the moment. Curiously there are two consistently failing tests indicating an issue with the media modal:
```
  ● Classic › should insert media, convert to blocks, and undo in one step

    TimeoutError: waiting for selector ".media-modal li[aria-label="5e354290-05fc-4699-8ee1-6fc80d864b34"]" failed: timeout 30000ms exceeded
```

and
```
  ● Classic › should insert media, convert to blocks, and undo in one step

    TimeoutError: waiting for selector ".media-modal li[aria-label="5e354290-05fc-4699-8ee1-6fc80d864b34"]" failed: timeout 30000ms exceeded
```

Possibly an issue uploading files?

**Comment by gziolo** [comment]
@talldan, is it using exactly the same version of WordPress to run e2e tests? Does it share the same configuration? I wouldn't be surprised if we discover some subtle differences.

**Comment by talldan** [comment]
I've rebased, there are more tests failing because `SCRIPT_DEBUG` seems to be `true` now in the test environment and several React errors that we know about are being thrown. Probably a good idea to fix those, so this might act as an impetus!

I won't be working for the rest of the week. If this seems like something someone would like to  take further feel free to push some commits to this branch or use this as a basic for a separate PR.

**Comment by talldan** [comment]
Oh, the issue with uploads may be a permissions issue. The previous implementation had these lines that I abruptly removed:
```
# Create the upload directory with permissions that Travis can handle.
mkdir -p wordpress/src/wp-content/uploads
chmod 767 wordpress/src/wp-content/uploads
```

From what I understand, with wp-env, the WordPress codebase is copied into a folder like `~/.wp-env/dd609e38eea1d81e91e8be063e246d80`, so it might be a challenge re-implementing this.

What do you think @noisysocks, does it sound like that might be the issue.

**Comment by noisysocks** [comment]
> I've rebased, there are more tests failing because SCRIPT_DEBUG seems to be true now in the test environment and several React errors that we know about are being thrown. Probably a good idea to fix those, so this might act as an impetus!

If you wanted to kick that can down the road you could set `SCRIPT_DEBUG` to `false` using something like `echo '{ "config": { "SCRIPT_DEBUG": false } }' > wp-env.override.json`.

> What do you think @noisysocks, does it sound like that might be the issue?

Hmm, are you able to see the issue with uploading media when you run E2E tests using `wp-env` locally? It's hard to say what's going on exactly without more info. If possible, though, I'd prefer to fix the issue in `wp-env` instead of using a `chmod` in the Travis script.

`wp-env start` will create a `wp-content/uploads` directory when it runs `docker-compose run cli wp core install`. The `cli` container runs this as the user with UID 33 which is `www-data` in the `wordpress` container. So, in theory, the `uploads` directory should be owned by `www-data` meaning the server in the `wordpress` container can write into it.

If we aren't able to reproduce this locally, we could explore getting extra debug info using one or more of these methods:

- Use `wp-env start --debug`, though I don't think it will show us anything useful in this case.
- Set `WP_DEBUG_LOG` to `true` in `.wp-env.override.json` so that WordPress writes all runtime errors to `wp-content/debug.log` and then have Travis `cat wp-content/debug.log` as its last step.
- Have Travis CI run `ls -la wp-content` so we can see what permissions `uploads` has.

**Comment by talldan** [comment]
Thanks for the info @noisysocks. e2e tests are now passing.

**Comment by gziolo** [comment]
There is still something wrong with PHPUnit tests.

We are getting closer to sunset `wp-scripts env` 🎉 

Great work you all 🥇

**Comment by talldan** [comment]
@gziolo One step closer! Not to see how we can progress #20900.

**Comment by talldan** [comment]
I notice there's also #22303 now, which might supersede this.

I rebased this anyway, and E2E tests are still passing, though PHP unit tests are still failing, which I found surprising.

**Comment by noahtallen** [comment]
Yeah that’s odd. PHP is already on wp-env and seems to be working on most other PRs!

File: packages/env/lib/commands/start.js
Comment (noahtallen) [inline packages/env/lib/commands/start.js:91]: Should we chmod more broadly to allow other things to have write access (e.g. chmod all of `wp-content`)? See some discussion around https://github.com/WordPress/gutenberg/pull/22454#issuecomment-631244208 in which a PHP test doesn't have access to what it wants

This may also remove the need for a different `chmod` I added in .travis.yml for the PHP test steps

**Comment by noahtallen** [comment]
Here's the phpunit error. It's actually something in wp-env:

```
> packages/env/bin/wp-env "start"

- 

✖ EACCES: permission denied, open '/home/travis/wp-env/ce047cf4da04af20491927334fb977c6/tests-WordPress/wp-config.php'

[Error: EACCES: permission denied, open '/home/travis/wp-env/ce047cf4da04af20491927334fb977c6/tests-WordPress/wp-config.php'] {

  errno: -13,

  code: 'EACCES',

  syscall: 'open',

  path: '/home/travis/wp-env/ce047cf4da04af20491927334fb977c6/tests-WordPress/wp-config.php'

}
```


This sounds awfully similar to this comment dealing with Ubuntu support: https://github.com/WordPress/gutenberg/issues/20180#issuecomment-590732930. I think the workaround _might_ just be to start the environment once (i.e. remove `wp-env start` from the phpunit test run in travis). But we still need to fix the underlying issue.

File: .travis.yml
Comment (talldan) [inline .travis.yml:56]: I had to also add `"WP_PHP_BINARY": "php"` here, as the default config was being overwritten by the .wp-env.override file. Using the override file like this won't be very maintainable.

Perhaps looking into command line options for overriding individual settings could be an option. e.g.:
```
npm run wp-env start --config SCRIPT_DEBUG=false
```

File: packages/env/lib/commands/start.js
Comment (talldan) [reply packages/env/lib/commands/start.js:91]: That seems like a good call. I can put together a separate PR to cover it.

File: .travis.yml
Comment (noahtallen) [reply .travis.yml:56]: we are also looking to add support for setting the config for different environments in .wp-env.json so that we can turn of SCRIPT_DEBUG etc for the tests environment without having to override anything.

**Comment by talldan** [comment]
> This sounds awfully similar to this comment dealing with Ubuntu support: #20180 (comment). I think the workaround might just be to start the environment once (i.e. remove wp-env start from the phpunit test run in travis). But we still need to fix the underlying issue.

Thanks for the troubleshooting @noahtallen, removing the other calls to `wp-env start` solved things.

Tests are passing now 🎉

**Review by noahtallen — APPROVED** [review]
Since the tests are passing, I'm not sure if we need to do anything else for this PR specifically :)

It uncovered some things that would be good for wp-env in the future though:
- fix linux environment issue
- .wp-env.json should handle config for different environments
- need better permissions in the wordpress install so that tests etc have write access to the filesystem

**Comment by talldan** [comment]
I've made a couple of issues:
- #22514 for test config
- #22515 for permissions

#20180 already covers linux env issues.

**Pull request merged**
